src/Microsoft.Dnx.Compilation.Abstractions/CompilationProjectContext.cs

```diff
@@ -24,6 +27,9 @@ public class CompilationProjectContext
             CompilationTarget target,
```

Comment:
devil's advocate - when a property has a getter and a setter you never know whether it is just for easy initialization or it actually being reset later. This can lead to wrong assumptions and bugs - where the code originally assumes it is only for initialization but then someone starts resetting the value. Also if you have many properties you probably have to rely on the IDE/Intellisense as well - only you can set them in any order...